src/main/java/com/bean00/Response.java

```diff
@@ -1,22 +1,99 @@
 package com.bean00;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class Response {
     private static final String PROTOCOL_VERSION = "HTTP/1.1";
+
+    private HashMap<String, Integer> headers = new HashMap<>();
     private int statusCode;
-    private String message;
+    private String body = "";
+    private int contentLength;
 
     public Response(int statusCode) {
         this.statusCode = statusCode;
-        this.message = Status.getMessage(statusCode);
+    }
+
+    public Response(int statusCode, byte[] rawBody) {
+        this(statusCode);
+        this.body = getBodyAsAString(rawBody);
+        this.contentLength = rawBody.length;
+        addHeaders();
     }
 
     @Override
     public String toString() {
-        return PROTOCOL_VERSION + " " + statusCode + " " + message + "\r\n";
+        String statusMessage = Status.getMessage(statusCode);
+
+        String response = buildResponse(statusMessage);
+
+        return response;
     }
 
     public int getStatusCode() {
         return statusCode;
     }
 
+    public String getBody() {
+        return body;
+    }
+
+    public int getHeader(String header) {
+        return headers.get(header);
+    }
+
+    private String getBodyAsAString(byte[] rawBody) {
+        String body = new String(rawBody);
+
+        return body;
+    }
+
+    private void addHeaders() {
+        headers.put("Content-Length", contentLength);
+    }
+
+    private String buildResponse(String statusMessage) {
+        String response = createInitialResponse(statusMessage);
+
+        response = addHeaders(response);
+
+        response = addBlankLine(response);
+
+        response = addBody(response);
+
+        return response;
+    }
+
+    private String createInitialResponse(String statusMessage) {
+        String response = PROTOCOL_VERSION + " " + statusCode + " " + statusMessage + "\r\n";
+
+        return response;
+    }
+
+    private String addHeaders(String response) {
+        if (!headers.isEmpty()) {
+            for (Map.Entry<String, Integer> header : headers.entrySet()) {
+                String headerString = header.getKey() + ": " + header.getValue() + "\r\n";
+                response += headerString;
+            }
+        }
+
+        return response;
```

Comment:
Instead of changing `response` and returning it, I think you can either make `response` an instance variable or just return a string that you can add up outside of this.

Reply:
Woops, looks like you two already caught this.

Reply:
Done